docs/v2_linter.md

```diff
@@ -0,0 +1,44 @@
+Linter to help migration to Conan v2
+====================================
+
+On our [path to Conan v2](v2_roadmap.md) we are leveraging on custom Pylint rules. This
+linter will run for every pull-request that is submitted to the repository and will
+raise some warnings and errors that should be addressed in order to migrate the
+recipes to Conan v2.
+
+It is important to note that these rules are targetting Conan v2 compatibility layer, their
+purpose is to fail for v1 syntax that will be no longer available in v2. Even if the syntax
+if perfectly valid in Conan v1, the recipe might fail here because it is not v2-complaint.
```

Comment:
```suggestion
if perfectly valid in Conan v1, the recipe might fail here because it is not v2-compliant.
```
```suggestion
if perfectly valid in Conan v1, the recipe might fail here because it is not v2-complaint.
```